### PR TITLE
Issue #11498: Added LITERAL_CATCH token support to WhitespaceAfter

### LINKED ISSUE #11498: WhitespaceAfter should support LITERAL_CATCH token
https://checkstyle.org/config_whitespace.html#WhitespaceAfter

I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```bash
/var/tmp $ javac CatchSpace.java
/var/tmp $

/var/tmp $ cat config.xml
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="WhitespaceAfter">
            <property name="tokens" value="LITERAL_CATCH"/>
        </module>
    </module>
</module>

/var/tmp $ cat CatchSpace.java
public class CatchSpace {
    public static void main(String[] args) {
        try {
            Integer.parseInt(args[0]);
        } catch(NumberFormatException e) {
            System.out.println("not a number");
        }
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.1-all.jar -c config.xml CatchSpace.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - Token "LITERAL_CATCH" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:476)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:403)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:330)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:189)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:126)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Token "LITERAL_CATCH" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.TreeWalker.registerCheck(TreeWalker.java:223)
        at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:133)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:471)
        ... 5 more
Checkstyle ends with 1 errors.
```

**Is your feature request related to a problem? Please describe.**
It would be nice if checkstyle could detect missing space after `catch` keyword.

**Describe the solution you'd like**
WhitespaceAfter could support new literal type - `LITERAL_CATCH`

**Additional context**
Google Java Style Guide requires whitespace separation of `(` from any reserved words
See https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace
>1. Separating any reserved word, such as `if`, `for` or `catch`, from an open parenthesis (`(`) that follows it on that line

#11489, #11493 and #11494 are similar requests for `try`/`finally`/`synchronized` keywords support

### PR DESCRIPTION
**Pull request by Binita-tech**
Resolves #11498 
Diff Regression config: https://gist.githubusercontent.com/Binita-tech/76be87d6c361ab5305863361aef3de15/raw/b207f4d59a809e58e13b45863e94479b7b3226e2/config.xml
Diff Regression patch config: https://gist.githubusercontent.com/Binita-tech/6fbe5f120efde70be0ad345079f86acc/raw/ef0b6c9270a7f0c03d82c67ce4ef9f2ed19be38f/patch_config.xml

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java [hunk 1]
```diff
@@ -100,6 +102,12 @@
  *
  *      try {} catch (Error e){} finally {} // OK
  *      try {} catch (Error e){} finally{} // violation ''finally' is not followed by whitespace'
+ *
+ *      try {} catch (Exception e){} // OK
+ *      try {} catch(Exception e){} // violation ''catch' is not followed by whitespace'
+ *
+ *      try {} catch (Error e){} finally {} // OK
+ *      try {} catch(Error e){} finally {} // violation ''catch' is not followed by whitespace'
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java:110]: Please remove such 2 examples with "finally "

File: src/xdocs/config_whitespace.xml [hunk 2]
```diff
@@ -2902,14 +2902,16 @@ class Bar {
...
+                  LITERAL_CATCH</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_DO">
                   DO_WHILE</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ELLIPSIS">
                   ELLIPSIS</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SWITCH">
                   LITERAL_SWITCH</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_TRY">
-                LITERAL_TRY</a>
+                  LITERAL_TRY</a>
```
Comment (romani) [inline src/xdocs/config_whitespace.xml:2914]: Please move update for indentation to separate commit of same PR.
Commit message "minor: fix indentation"

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java (hunk 1 above)
Comment (Binita-tech) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java]: Done

File: src/xdocs/config_whitespace.xml (hunk 2 above)
Comment (Binita-tech) [reply src/xdocs/config_whitespace.xml:2914]: Done

**Review by romani — APPROVED** [review]
Ok to merge

**Review by nick-mancuso — CHANGES_REQUESTED** [review]
@Binita-tech please generate check regression report. You will need to use a patch config, see instructions at https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#report-generation

You can see an example at https://github.com/checkstyle/checkstyle/pull/11491

**Comment by Binita-tech** [comment]
Done @nick-mancuso

**Comment by nick-mancuso** [comment]
GitHub, generate report

cc @Binita-tech

**Comment by nick-mancuso** [comment]
GitHub, generate report

**Comment by nick-mancuso** [comment]
GitHub, generate report

**Comment by Binita-tech** [comment]
GitHub, generate report

**Comment by Binita-tech** [comment]
GitHub, generate report

**Comment by Binita-tech** [comment]
GitHub, generate report

**Comment by Binita-tech** [comment]
GitHub, generate report

**Comment by Binita-tech** [comment]
GitHub, generate report

**Comment by nick-mancuso** [comment]
@Binita-tech please try to reformat PR description and make sure that there is only a single space between `:` and link to configs.

**Comment by Binita-tech** [comment]
GitHub, generate report

**Comment by Binita-tech** [comment]
Done @nick-mancuso

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java [hunk 3]
```diff
@@ -29,7 +29,7 @@ public void check3(int... a) {
 
     public void check4() throws java.io.IOException {
         try (java.io.InputStream ignored = System.in) {}
-        try {} catch(Exception e){}
+        try {} catch (Exception e){}
```
Comment (nick-mancuso) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java:32]: This change should be in main commit `Issue #..`, since it is related to your changes.

**Review by nick-mancuso — CHANGES_REQUESTED** [review]
Item:

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java (hunk 3 above)
Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java:32]: Yes, this commit should be only for indentation

Comment (Binita-tech) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java:32]: Do I have to make changes? This indentation fix was not related to my issue.

Comment (romani) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java:32]: Do change, but do this change in another commit

Comment (nick-mancuso) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java:32]: > Do I have to make changes? This indentation fix was not related to my issue.

This became a violation when you added `LITERAL_CATCH` token to this check, so it is part of this issue :)

Comment (Binita-tech) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java:32]: Done

**Review by nick-mancuso — APPROVED** [review]

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java (hunk 3 above)
Comment (nick-mancuso) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java:32]: Resolved.

**Review by pbludov — APPROVED** [review]

**Comment by Binita-tech** [comment]
If possible, please merge soon otherwise it would create merge conflict again by another PR of similar changes.
